tests/components/ecovacs/test_binary_sensor.py

```diff
@@ -0,0 +1,49 @@
+"""Tests for Ecovacs binary sensors."""
+
+from deebot_client.event_bus import EventBus
+from deebot_client.events import WaterAmount, WaterInfoEvent
+import pytest
+from syrupy import SnapshotAssertion
+
+from homeassistant.components.ecovacs.controller import EcovacsController
+from homeassistant.const import STATE_OFF, STATE_UNKNOWN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+
+from .util import notify_and_wait
+
+pytestmark = [pytest.mark.usefixtures("init_integration")]
+
+
+async def test_mop_attached(
+    hass: HomeAssistant,
+    device_registry: dr.DeviceRegistry,
+    entity_registry: er.EntityRegistry,
+    controller: EcovacsController,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test mop_attached binary sensor."""
+    assert (state := hass.states.get("binary_sensor.ozmo_950_mop_attached"))
+    assert state.state == STATE_UNKNOWN
+
+    assert (entity_entry := entity_registry.async_get(state.entity_id))
+    assert snapshot == entity_entry
```

Comment:
With snapshot tests I want to recommend giving your entity and state snapshots a name. Because If you add more entities in the future, the list can get messed up (as order changes).

In this case you only have 1 entity to test, but for other platforms, please checkout tests like this:
https://github.com/home-assistant/core/blob/0d8afc72c2304d21d1286e93ce675019acd4d840/tests/components/glances/test_sensor.py#L13-L29

Reply:
Done it also for the binary sensor